.dapper

```diff
@@ -0,0 +1,100 @@
+#!/bin/sh
+
+params="$(getopt -o f:km:OC:sdqXS:hv -l file:,socket,mode:,no-out,build,directory:,shell,debug,quiet,keep,no-context,mount-suffix,target,help,version --name "$0" -- "$@")"
+eval set -- "$params"
+
+file=Dockerfile.dapper
+socket=false
+dockerargs=
+
+while true
+do
+    case "$1" in
+	--file|-f)
+	    file="$2"
+	    shift 2
+	    ;;
+	--socket|-k)
+	    socket=true
+	    shift
+	    ;;
+	--mode|-m)
+	    # Ignored, we only support bind
+	    shift 2
+	    ;;
+	--no-out|-O)
+	    # Ignored, we only support bind
+	    shift
+	    ;;
+	--build)
+	    # Not supported
+	    echo "$0 doesn't support --build yet"
+	    exit 1
+	    ;;
+	--directory|-C)
+	    cd "$2" || exit
+	    shift 2
+	    ;;
+	--shell|-s)
+	    command=bash
+	    dockerargs="${dockerargs} -it"
+	    DAPPER_ENV="${DAPPER_ENV} TERM"
+	    shift
+	    ;;
+	--debug|-d)
+	    shift
+	    set -x
+	    ;;
+	--quiet|-q|--keep|--no-context|-X|--help|-h)
+	    # Ignored
+	    shift
+	    ;;
+	--version|-v)
+	    echo Submariner Dapper
+	    exit 0
+	    ;;
+	--mount-suffix|-S)
+	    suffix=":$2"
+	    shift 2
+	    ;;
+	--)
+	    shift
+	    break
+	    ;;
+	*)
+	    echo "$0 doesn't support $1" >&2
+	    exit 1
+	    ;;
+    esac
+done
+
+[ -n "$command" ] && set -- "$command"
+
+gitid="$(git branch --show-current | tr / _)"
```

Comment:
I'm getting this on OSX

```
./.dapper  -m bind  -- make --debug=b clusters
error: unknown option `show-current'
usage: git branch [<options>] [-r | -a] [--merged | --no-merged]
   or: git branch [<options>] [-l] [-f] <branch-name> [<start-point>]
   or: git branch [<options>] [-r] (-d | -D) <branch-name>...
   or: git branch [<options>] (-m | -M) [<old-branch>] <new-branch>
   or: git branch [<options>] (-c | -C) [<old-branch>] <new-branch>
   or: git branch [<options>] [-r | -a] [--points-at]
   or: git branch [<options>] [-r | -a] [--format]

Generic options
    -v, --verbose         show hash and subject, give twice for upstream branch
    -q, --quiet           suppress informational messages
    -t, --track           set up tracking mode (see git-pull(1))
    -u, --set-upstream-to <upstream>
                          change the upstream info
    --unset-upstream      Unset the upstream info
    --color[=<when>]      use colored output
    -r, --remotes         act on remote-tracking branches
    --contains <commit>   print only branches that contain the commit
    --no-contains <commit>
                          print only branches that don't contain the commit
    --abbrev[=<n>]        use <n> digits to display SHA-1s
```

But trying to update my git

Reply:
ok, new git version, and got past that:
```
./.dapper  -m bind  -- make --debug=b clusters
Sending build context to Docker daemon  1.414MB
Step 1/8 : ARG BASE_BRANCH
Step 2/8 : FROM quay.io/submariner/shipyard-dapper-base:${BASE_BRANCH}
 ---> 46a4c9bc7227
Step 3/8 : ARG PROJECT
 ---> Using cache
 ---> 308459f85479
Step 4/8 : ENV DAPPER_ENV="QUAY_USERNAME QUAY_PASSWORD CLUSTERS_ARGS DEPLOY_ARGS CLEANUP_ARGS E2E_ARGS RELEASE_ARGS MAKEFLAGS FOCUS SKIP PLUGIN E2E_TESTDIR"     DAPPER_SOURCE=/go/src/github.com/submariner-io/${PROJECT} DAPPER_DOCKER_SOCKET=true
 ---> Using cache
 ---> 7bf5b15bcbcc
Step 5/8 : ENV DAPPER_OUTPUT=${DAPPER_SOURCE}/output
 ---> Using cache
 ---> 9c5a1a920a32
Step 6/8 : WORKDIR ${DAPPER_SOURCE}
 ---> Using cache
 ---> d3a7485f550f
Step 7/8 : ENTRYPOINT ["/opt/shipyard/scripts/entry"]
 ---> Using cache
 ---> 07702556c379
Step 8/8 : CMD ["sh"]
 ---> Using cache
 ---> 13913dd1a649
Successfully built 13913dd1a649
Successfully tagged shipyard:our-own-dapper
sed: 1: "/"DAPPER_ENV=/{s/[[:spa ...": extra characters at the end of q command
sed: 1: "/"DAPPER_SOURCE=/{s/[[: ...": extra characters at the end of q command
sed: 1: "/"DAPPER_DOCKER_SOCKET= ...": extra characters at the end of q command
sed: 1: "/"DAPPER_RUN_ARGS=/{s/[ ...": extra characters at the end of q command
docker: invalid reference format: repository name must be lowercase.
See 'docker run --help'.
```

Reply:
installing gnu-sed seems to fix this. Continuing with the tests.

Reply:
Ah yes, `git branch --show-current` is somewhat recent, I’ll replace it with `git symbolic-ref --short HEAD`.